zerver/actions/create_realm.py

```diff
@@ -232,6 +234,14 @@ def do_create_realm(
 
         create_system_user_groups_for_realm(realm)
 
+        # We create realms with all authentications methods enabled by default.
+        RealmAuthenticationMethod.objects.bulk_create(
+            [
+                RealmAuthenticationMethod(name=backend_name, realm=realm)
+                for backend_name in all_implemented_backend_names()
+            ]
+        )
```

Comment:
Hmm, how does this interact with things like SAML backends limited to certain subdomains? Maybe it still makes sense to do this, but it probably deserves some thought; I could imagine having some form of "which are enabled by default" feature.

Reply:
This doesn't interact with SAML in any new way right now  - it's just just how it was with the BitField, which just had a general "SAML" flag. Now it just has a possible `name="SAML"` row. Which specific IdPs are enabled is still enforced via the configuration inside `SOCIAL_AUTH_SAML_ENABLED_IDPS`. But indeed, with the BitField gone, this new design should be much more naturally extandable to support controling enabled IdPs here. Just not sure if we actually would want that? Because on Cloud, we don't want customer X to have customer's Y SAML IdP showing up in their organization's auth settings UI at all. But it would make more sense on self-hosted servers I guess

Reply:
Yeah probably best to leave this as-is for this migration; trying to make that fancier could make sense but it'd be a follow-up.